MRG: Reconfigure website

Eric89GXL opened the pull request:
Closes #4249.

Eric89GXL commented:
https://4581-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

choldgraf commented:
Can we call it "All of the new things that are coded in MNE-python"?

JK

I like the new layout...one quick thought, there's some weird stuff going on when you make the screen super wide:

![image](https://cloud.githubusercontent.com/assets/1839645/25821382/bcc2d36a-33e9-11e7-879f-bdf8acadf3b2.png)

agramfort commented:
can you insert an INRIA logo:

https://www.inria.fr/var/inria/storage/images/medias/inria/images-corps/logo-inria-sans-signature-couleur/404509-2-fre-FR/logo-inria-sans-signature-couleur.jpg

while you're at it?

choldgraf commented:
what about BIDS? :)

https://bids.berkeley.edu/sites/all/themes/bids/logo.png

Eric89GXL commented:
Sure. It looks like it auto-shrinks if the banner becomes too large anyway

mmagnuski commented:
Works for me well for the normal window size, but the rescaling is weird when the window gets smaller (but it is not a new problem):
<img width="585" alt="mne page pic 01" src="https://cloud.githubusercontent.com/assets/8452354/25829689/b183d3a8-344f-11e7-9b11-826aadaa4f9f.PNG">

Eric89GXL commented:
Logos updated with Inria and BIDS. Let's fix the UI bugs in another PR when someone gets sufficiently motivated.

choldgraf commented:
that's fine with me!

Eric89GXL commented:
Ready to go from my end

https://4583-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

agramfort commented:
I personally think that to make the what's new page more visible we lost
many direct links previously on the right column (link to mailing list, how
to cite, etc).

I think I prefer the current layout...

Eric89GXL commented:
I think that the landing page should really be about getting people interested in / introduced to the package, so adding more links actually ends up detracting from the page. I can add them back in if they are seen as essential top-level links, but I'd rather make them more prominent on the Documentation page instead.

choldgraf commented:
So, I apologize ahead of time because this is potentially creating more work for folks, but I'm gonna throw out a thought:

Why don't we re-work MNE's front page? Hear me out...

The current website of MNE is *great*, but I think we can do a much better job of prioritizing and organizing information on the front page. I propose that we model this off of the scikit-learn front page. We don't need to do everything the same, but the SKL frontpage is really about getting straight to the point: what does this package let you do? They've got visuals on the frontpage which are great for drawing people in, and links to immediately-relevant information for new (or old) users.

E.g., this is what pops up on my screen when I go to the SKL website:

![image](https://cloud.githubusercontent.com/assets/1839645/25908314/bbf06170-355e-11e7-8618-47b309fc85fd.png)

* tons of information about what scikit-learn does
* visuals that look nice and give a feeling for what scikit-learn does
* lots of larger-font text that pops out
* much fewer "small" text
* almost all pieces of information have a link to something else on the website.
* funding groups at the bottom

Compare that with the MNE

![image](https://cloud.githubusercontent.com/assets/1839645/25908292/ab3778e6-355e-11e7-95b6-609edc38ffdd.png)

* ~50% just MNE branding + funding information. Neither of which is useful to people.
* The only information that tells us what MNE does is that bullet list of HTML text w/o any links
* No headers etc that let users visually group information more easily
* 20 lines of code that isn't that useful unless you know what it actually produces.

What I'd propose is that we:

1. Replace the "20 lines of code" with an image w/ a caption that says "reproduce this figure with 20 lines of code" that is clickable and takes you to an example.
2. Even better, have a sliding set of figures, each of which can be recreated w/ < 20 lines of code.
5. Make the "topics" list as a grid of topics w/ larger font, each item in the grid is clickable and takes you to the documentation for that topic. Maybe one button per section in the tutorial?
6. Switch the twitter feed to something other than MNE-python, since that's just a bot that parrots github activity. better would be MNE-news.
7. Move all of the logos + funding information to the bottom.

I'm not saying that we should do this in the current PR, but what do people think about this? I can open an issue (or a PR) to look into this if folks are friendly to it.

Eric89GXL commented:
+100 from me

You can start with this branch as your basis and open a new PR. At least you'll get the updated logos :)

choldgraf commented:
Looks like this is where sklearn is doing their carousel code:

https://github.com/scikit-learn/scikit-learn/blob/master/doc/themes/scikit-learn/layout.html#L136

Think we could put together a list of images in the examples that would be worth showing off, and that highlight the diversity of MNE functionality?

Eric89GXL commented:
Yes, we can even start from the list of bullet points on the main page. But we need to see if others are on board first.

choldgraf commented:
Agreed - I'll hold off on opening a PR until I get a plus 1 from the grand pooh-bah at least ;-)

agramfort commented:
can you do a mockup with a slide on google doc so we can suggest / comment?

choldgraf commented:
@agramfort something like this: https://docs.google.com/presentation/d/1LJIh_mL0FhYa-nWBsPVHJ8Pj9NBUXCUTIVf13GkuK_A/edit?usp=sharing

Eric89GXL commented:
LGTM

teonbrooks commented:
@choldgraf I like it :)

agramfort commented:
I like it too. I'd like also to have quick access to what we presently call
"Help" mailing list and how to contacts ie what sklearn calls "Community"

Eric89GXL commented:
But let's not take up too much space with it :) Maybe we can have a "Community" button that has a drop-down with links to github/gitter/mailing list. @choldgraf sounds like you have a green light!

jasmainak commented:
It's cool! But I'm concerned that there is no link to "Download" or "Installation" on the home page. It's usually the first thing I look for when I visit a software page ...

Eric89GXL commented:
@jasmainak there is a "Get started" as the first option in the top title bar, is it not sufficient?

jasmainak commented:
Maybe :) "Get started" is a little obscure to me. I'd prefer "Installation".

Eric89GXL commented:
It's a bit more than just installation, but I can live with it. Though I'd prefer "Install" so it's shorter in the top bar but still conveys the same meaning.

jasmainak commented:
yeah but I don't understand why installation instructions are nested two levels deep. [This page](https://martinos.org/mne/stable/install_mne_python.html) and [this page](https://martinos.org/mne/stable/install_mne_c.html) should really be linked from the home page.

Eric89GXL commented:
We don't want to risk cluttering the landing page. Let's do Chris's
modification since we've converged on that, and you can follow up with a PR
proposal to change the link structure (since it's mostly an orthogonal
change)

choldgraf commented:
@Eric89GXL I spent a bit of time today to try and prototype what this'd look like in the codebase. I just pushed these changes to your fork in case you'd like to iterate on this a little bit as well. Does that sound good?

Here's where it stands right now:

http://predictablynoisy.com/mne-python/

As I said, a long way to go but I think it should be easier to iterate from here. I'm also linking to thumbnails from random examples currently because I didn't want to have to build all of the examples.

Eric89GXL commented:
Okay, made some changes:

https://4633-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

Notably:

1. Aesthetic changes relative to what @choldgraf had.
2. Fixed the quick-link buttons @choldgraf had put in to properly unfold the correct documentation section (to satisfy @kingjr's desire to link to the doc directly on the homepage).
3. Added `whats_new` and other important quick-links to footer (to satisfy @agramfort).
4. Changed "Getting started" to "Install" (to satisify @jasmainak).

See what you think.

kingjr commented:
Very nice! I really like this new organization, it's much more functional. 

A couple of minor remarks
* can you add a `v` mark on the main topics to suggest that it's clickable
* I'm not a big fan of the image selection on the front page. Several are pixelated. If possible I would change the frontpage layout such that we don't need to scroll down for a 1024*768 screen.
* What is the meaning of the logos related to Speed, usability etc?

Again, very nice!

Eric89GXL commented:
> can you add a v mark on the main topics to suggest that it's clickable

I'm not sure what you mean.

> I'm not a big fan of the image selection on the front page. Several are pixelated. If possible I would change the frontpage layout such that we don't need to scroll down for a 1024*768 screen.

I'll try to trim a little bit of space.

> What is the meaning of the logos related to Speed, usability etc?

Just some eye candy, but clearly they didn't translate properly. I'll try to find some more relevant ones.

choldgraf commented:
Yo! Some thoughts:

+1s
* I really like the text you've put into the examples and the general look of the slider now.
* I like the idea of the little callout things to the right. (though I wonder if we should come up with something other than "reliability" as most users probably don't think about this kind of thing. It feels a bit more like it's catering to devs rather than avg scientists. Why not something like "Flexibility" and the sub-text would describe how it is meant to handle many workflows and types of data.)
* The footer section w/ the institutions is nice! 
* I also like the formatting of the financial support section.

**Comments / suggestions**
* The number of dots in the slider doesn't match the number of slider items.
* The whitespace around MNE looks a little awkward to me. What if we went with a fixed-width layout that kept the look of MNE consistent across screen sizes. Maybe 800-900px? nts / suggestions
* I'd be a fan of making those call-out points to the right be the same height as the MNE logo, and the text below that's more verbose would span the whole width of the above two elements. Or to have those bullet points scale so their height is always the same as the logo + text height. (or as mentioned above, we just go with fixed width so it always looks the same).
* The slider is below-the-fold on my screen. Perhaps we can shrink the whitespace at the top. At the least I want to make sure that text and "try it" button of each slider image is visible on a ~13" screen.
* I'm not sure how I feel about the buttons changing their width based on the length of the text inside. Were you not a fan of fixed-width buttons?

I think that most of the visual components are here...we just need to improve the layout / spacing a bit and make sure that the most important information is prioritized in terms of what the user sees.

choldgraf commented:
ah, and to @kingjr 's point about the image slider, I think we should select examples we want to have in the slider and make sure that those examples output images that are higher-res than normal.

Eric89GXL commented:
> I'm not sure how I feel about the buttons changing their width based on the length of the text inside. Were you not a fan of fixed-width buttons?

They made the layout harder to follow IMO. Feel free to play with it locally

choldgraf commented:
I agree w/ @Eric89GXL about not getting everything perfect. It will be easy to iterate on specific examples etc once the main machinery is there. I think this PR should focus on layout / broad design principles.

But in another thread I agree we should discuss specific examples we want to show (e.g. there should definitely be a `joint` plot displayed or something like this). We should think "for the most common use-cases of analysis in EEG/MEG/ECoG, what is the single coolest-looking viz we can come up with?"

Eric89GXL commented:
> I'd be a fan of making those call-out points to the right be the same height as the MNE logo, and the text below that's more verbose would span the whole width of the above two elements. Or to have those bullet points scale so their height is always the same as the logo + text height. (or as mentioned above, we just go with fixed width so it always looks the same).

I tried a ton of iterations (this included, I think?) and the current one seemed to work best. Feel free to try something else locally.

I'll see if I can make the images a bit nicer, but if I can't I'd say let's do that in a follow-up PR. (I don't mind the light pixelation/upsampling personally.)

I'll take a stab at the height/whitespace, though, since it would be nice to fit on a single page.

choldgraf commented:
My 2 cents, this:

![image](https://cloud.githubusercontent.com/assets/1839645/26212322/0a8c2e44-3baa-11e7-894c-e42ccd74d7bc.png)

Is more in-line with modern web design principles :-)

Eric89GXL commented:
That's fine, I just want them to span the horizontal space. So feel free to push that button mod. Or probably better to make a PR so we don't conflict (I'm going to tackle whitespace now).

choldgraf commented:
but TBH I think that a lot of my comments about the call-out buttons etc would take a backseat if we could figure out a way to reduce the whitespace around the logo and to make the slider more pronounced above the fold

choldgraf commented:
> Or probably better to make a PR so we don't conflict (I'm going to tackle whitespace now).

+1 , I'll wait until this gets merged and then tackle some of the smaller visual stuff

choldgraf commented:
(and apologies if any of this is annoying feedback...I tend to have strong opinions about design :-P)

Eric89GXL commented:
Okay I squeezed some space, tweaked some names/titles, and made the images less pixelated (by shrinking the carousel):

https://4640-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

Good enough for now, or more tweaks?

choldgraf commented:
looks a lot better IMO. The only thing that bugs me now is the buttons on the right being right-justified, and the funding text below being left-justified...do the buttons look weird if they're left-justified?

Eric89GXL commented:
> do the buttons look weird if they're left-justified?

Yeah. You can test these things locally without needing to re-build the site. Even directly in chrome you can right-click -> "inspect" and then change CSS properties.

choldgraf commented:
what about:

Eric89GXL commented:
I don't like it as much personally

Eric89GXL commented:
I tried it again, and I guess the left-align is *okay*, so if you really dislike the right I can change it

choldgraf commented:
Maybe the thing that's bugging me is the checkbox - I think this is awkward w/ a right-justification, and also it makes me want to click it bc it looks like something interactive. What about just:

![image](https://cloud.githubusercontent.com/assets/1839645/26216414/d98def94-3bb8-11e7-9e2a-39e20795545b.png)

Eric89GXL commented:
I like having the icons not just for eye-candy, but also as a visual anchor / bullet-type behavior.

Eric89GXL commented:
If you want to iterate on this, we should do it on Gitter to avoid flooding GitHub notifications.

choldgraf commented:
(in gitter now)

agramfort commented:
@Eric89GXL did you try to put the carousel on the right of the MNE logo? it would allow to have the content boxes visible without scrolling down.

dengemann commented:
Looks really nice, awesome work Chris & Eric! Almost uncannily professional
for academic software :) Can we put back some piece of demo code below to
expose a bit our API abstractions?
I agree we need to select the examples we want to advertise. The source
permutation clustering stats for example might not be among them.


On Fri, May 19, 2017 at 9:44 AM Alexandre Gramfort <notifications@github.com>
wrote:

> @Eric89GXL <https://github.com/eric89gxl> did you try to put the carousel
> on the right of the MNE logo? it would allow to have the content boxes
> visible without scrolling down.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/4250#issuecomment-302633439>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB0fiq__CCWXXNLCRzjx1CreiVPt8F__ks5r7UhxgaJpZM4NUWvw>
> .
>

Eric89GXL commented:
We initially had it to the right but things were crowded.

I don't like having the code because it clutters the page. Ideally things
should fit on a single page without scrolling. The more things we add, the
harder it is to get where people want. People agreed on the initial mockup
that didn't have it, so I assume we already otherwise have consensus to get
rid of it.

Maybe instead I can make an advertisement by making the first carousel page
say you can go from raw to dSPM in 20 lines?

As per the previous discussion, can we can refine examples in a dedicated
follow-up PR? This one is focused on layout as much as possible.

dengemann commented:
Yes. No reason to pur everything in 1 PR.

On Fri, 19 May 2017 at 13:16, Eric Larson <notifications@github.com> wrote:

> We initially had it to the right but things were crowded.
>
> I don't like having the code because it clutters the page. Ideally things
> should fit on a single page without scrolling. The more things we add, the
> harder it is to get where people want. People agreed on the initial mockup
> that didn't have it, so I assume we already otherwise have consensus to get
> rid of it.
>
> Maybe instead I can make an advertisement by making the first carousel page
> say you can go from raw to dSPM in 20 lines?
>
> As per the previous discussion, can we can refine examples in a dedicated
> follow-up PR? This one is focused on layout as much as possible.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/4250#issuecomment-302676649>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB0fitkT7tTbXMuWEkOPsqcTSI3y08G3ks5r7XolgaJpZM4NUWvw>
> .
>

agramfort commented:
On Fri, May 19, 2017 at 1:16 PM, Eric Larson <notifications@github.com>
wrote:

> We initially had it to the right but things were crowded.


can I see screenshot? sorry to be late again on the discussion ... :(

Eric89GXL commented:
Had to re-do it. But it's something like this:

![screenshot from 2017-05-19 09-58-47](https://cloud.githubusercontent.com/assets/2365790/26250970/c52e98b6-3c79-11e7-8aae-4b43c9830954.png)

This is what is currently proposed (same window size):

![screenshot from 2017-05-19 10-00-12](https://cloud.githubusercontent.com/assets/2365790/26251056/0838539a-3c7a-11e7-8a94-e23e4fb649b3.png)

I don't like the gallery-right layout as much, it doesn't flow as well to my eye. If the concern is sizing, I could make some of the current elements smaller.

agramfort commented:
what if you make the 8 darks boxes bigger centered mimicking the
scikit-learn layout?

http://scikit-learn.org/stable/

Eric89GXL commented:
Not sure if you're talking about the current layout or the left-right one. I'm also not sure what exactly you mean by mimicking their layout (break out the buttons into info boxes? what do you mean by centered, do you mean three columns?).

But here are both with bigger buttons at least:

![screenshot from 2017-05-19 10-39-32](https://cloud.githubusercontent.com/assets/2365790/26252911/901bc0f8-3c7f-11e7-9bc6-1d4ee35be822.png)

Current:

![screenshot from 2017-05-19 10-44-12](https://cloud.githubusercontent.com/assets/2365790/26253086/1f66760e-3c80-11e7-8bce-858506337e5a.png)

choldgraf commented:
Or do you mean something more like this @agramfort ?

http://predictablynoisy.com/mne-python/

This was one of the earlier versions of the PR but we had modified it to the current version because we didn't want it to be too cluttered.

agramfort commented on doc/_static/style.css:
```diff
@@ -167,8 +198,90 @@ dt:target code {
+    font-size: 16px;
+}
+.tagline {
+  font-size: 24px !important;
```
can you use:

.tagline {
  font-size: 22px !important;
  text-align: center;
}

agramfort commented:
also we need to extend the bootstrap navbar breakdown width:

https://stackoverflow.com/questions/19827605/change-bootstrap-navbar-collapse-breakpoint-without-using-less

also can you make the fonts in the buttons bigger?

also when making the window less wide the interesting buttons appear at the very end of the page. I think they should appear maybe just after the logo so no need to scroll down on a phone or tablet.

sorry for the late feedback...

thanks a ton @Eric89GXL and @choldgraf for taking a stab at this. This is very very useful.

Eric89GXL commented:
> .tagline {
> font-size: 22px !important;
> text-align: center;
> }

I like the size better, but this is too much centering IMO.

I'll look into the other fixes, too, and send an image soon I hope.

Eric89GXL commented:
> also when making the window less wide the interesting buttons appear at the very end of the page. I think they should appear maybe just after the logo so no need to scroll down on a phone or tablet.

I tried this and it doesn't look good. I think that such a narrow case is going to be far less common than a fuller-width case. I'd rather not sacrifice the vast-majority case to make the rare case look better.

Instead I think I found a different way to display things that still stacks decently on smaller widths:

https://4661-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

choldgraf commented:
Can we have a short skype on this sometime this week? Designing by committee is already hard enough, and doing so asynchronously is even more difficult.

my 2 cents on the buttons is that they should go under the image slider, since people generally respond better to pictures it should be given priority in the screen real-estate.

Also @Eric89GXL there's a type in the "receptive field" description for the slider. Just FYI.

agramfort commented:
beautiful !

+1 for merge.

@choldgraf I give you the privilege to merge this !

agramfort commented:
:beers: !

teonbrooks commented:
looks great! nice work!! 🍻
